Sources/SwiftDocC/Semantics/Metadata/Availability.swift

```diff
@@ -77,6 +74,7 @@ extension Metadata {
                 case .iOS: return "iOS"
                 case .watchOS: return "watchOS"
                 case .tvOS: return "tvOS"
+                case .any: return "*"
```

Comment:
It would look something like the screenshot below:

<img width="743" alt="Screenshot 2024-03-22 at 5 33 26 PM" src="https://github.com/apple/swift-docc/assets/26250654/c8aa1081-9a3b-4f25-81e1-e6322324470a">

Reply:
I don't know how this is meant to work but I don't think we'd want this to display `*` here. 

It's a bit odd that the wildcard specifies a version at all because that version may not align across the various platforms.

Reply:
If that's the case what would be the correct representation to say that it is deprecated on all platforms?

Where as a specific case of deprecation might look like this for example:

<img width="744" alt="Screenshot 2024-03-28 at 10 10 36 PM" src="https://github.com/apple/swift-docc/assets/26250654/cc7000ad-657b-4aed-9a33-902a86de6b24">

Reply:
I would probably say that the correct way would be to list the deprecation on platform where the symbol is known to be available. 

However, like I said above, it's a bit odd that we allow specifying a deprecation version for all platforms since the different platforms don't need to align their versions and in practice don't do so.

Reply:
Ah that makes sense, I'll update the PR as soon as possible with removing the `*`. And thus aligning with screenshot sent below/ in the previous comment:

<img width="744" alt="Screenshot 2024-03-28 at 10 10 36 PM" src="https://github.com/apple/swift-docc/assets/26250654/cc7000ad-657b-4aed-9a33-902a86de6b24">

Reply:
Should be fixed now